dll/win32/shell32/COpenWithMenu.cpp

```diff
@@ -363,7 +380,10 @@ BOOL COpenWithList::LoadInfo(COpenWithList::SApp *pApp)
     /* Query name */
     swprintf(wszBuf, L"\\StringFileInfo\\%04x%04x\\FileDescription", wLang, wCode);
     if (VerQueryValueW(pBuf, wszBuf, (LPVOID *)&pResult, &cchLen))
+    {
         StringCchCopyNW(pApp->wszName, _countof(pApp->wszName), pResult, cchLen);
+        success = cchLen > 1;
+    }
```

Comment:
I wonder whether you could rearrange this block as follows:
```suggestion
    success = VerQueryValueW(pBuf, wszBuf, (LPVOID *)&pResult, &cchLen) && (cchLen > 1);
    if (success)
        StringCchCopyNW(pApp->wszName, _countof(pApp->wszName), pResult, cchLen);
```
in case retrieving the name is what really differentiates between a successful or a failed call (independently of whether the StringCchCopyNW call succeeded or not).

Reply:
Not checking StringCchCopyNW because we don't care if the name fits in the provided buffer (250 characters) or if it is truncated (SUCCEEDED will not help us) and the proper logic is not worth it, we will end up with something usable either way.

Reply:
You mean we don't care because that name is something that will be only displayed, so it would be just a minor inconvenience if it gets truncated?